engine/graphics/src/vulkan/graphics_vulkan.cpp

```diff
@@ -1469,7 +1473,12 @@ namespace dmGraphics
     {
         assert(size > 0);
         DeviceBuffer* buffer = new DeviceBuffer(VK_BUFFER_USAGE_INDEX_BUFFER_BIT);
-        DeviceBufferUploadHelper((VulkanContext*) context, data, size, 0, (DeviceBuffer*) buffer);
+
+        if (size > 0)
+        {
+            DeviceBufferUploadHelper((VulkanContext*) context, data, size, 0, buffer);
+        }
```

Comment:
We get a hard crash if we try to upload a zero buffer. The vertex buffer equivalent function does this as well